### PR TITLE
Enable all the Kotlin multiplatform targets

### PR DESCRIPTION
**Pull request by nicolasfara**
This PR enables all the available targets for a Kotlin multiplatform project.

The CI runs on the `macos` machine since we can cross-compile for all the other targets from that host.

## Limitations

A workaround is needed to enable the following targets: `linuxArm64`, `linuxArm32Hfp`, and `mingwX86`.
The workaround force the place of the `kotest` libraries to be not in the `commonTest` source folder since all the targets depends on it.

In this way, it is impossible to use `kotest` inside the `commonTest` source folder.
In order to use `kotest`, a test class should be specified for each specific target (`jsTest`, `jvmTest`, and `nativeTest`).

### CODE REVIEW

**Comment by DanySK** [comment]
We lose the ability to test on Win and Linux, though.
I propose a different approach: we distinguish between two build types: tests, where we pull in kotest and only enable the kotest-supported platforms, and assemble, in which we enable everything and require macos.
We can use a Gradle property in CI to switch the behaviour, have build running on all three platforms, and another task performing a delivery if tests succeed.

Does it work?

**Comment by nicolasfara** [comment]
I've restored all the source folders as before. The Gradle property `releaseStage` enable all the non-compatible `kotest` platforms and the Gradle property is set in CI during the release job on macOS.
Is this a valid solution? Or do you have more suggestions to make it better?

**Review by DanySK — APPROVED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -23,6 +23,16 @@ repositories {
     mavenCentral()
 }
 
+val binarySetup: KotlinNativeTarget.() -> Unit = {
```
[the remark this answers is not in the document]
Comment (nicolasfara) [reply build.gradle.kts]: Fixed in a98cf69

**Comment by DanySK** [comment]
configured for automerge with rebase and auto-rebase-update.

**Pull request merged**